accounting/accounting_cli_functions.py

```diff
@@ -13,10 +13,170 @@
 import argparse
 import time
 import sys
+import pwd
+import json
 
 import pandas as pd
 
 
+def count_ranks(ranks):
+    if "-" in ranks:
+        ranks_count = ranks.replace("-", ",").split(",")
+        return int(ranks_count[1]) - int(ranks_count[0]) + 1
+    else:
+        return int(ranks) + 1
+
+
+def get_username(userid):
+    try:
+        return pwd.getpwuid(userid).pw_name
+    except KeyError:
+        return str(userid)
+
+
+def get_uid(username):
+    try:
+        return pwd.getpwnam(username).pw_uid
+    except KeyError:
+        return str(username)
+
+
+def write_records_to_file(job_records, output_file):
+    op = open(output_file, "a+")
+    op.write("UserID,Username,JobID,T_Submit,T_Run,T_Inactive,Nodes,R\n")
+    for record in job_records:
+        op.write(
+            str(record.userid)
+            + ","
+            + str(record.username)
+            + ","
+            + str(record.jobid)
+            + ","
+            + str(record.t_submit)
+            + ","
+            + str(record.t_run)
+            + ","
+            + str(record.t_inactive)
+            + ","
+            + str(record.nnodes)
+            + ","
+            + str(record.R)
+            + ",\n"
+        )
```

Comment:
I'd recommend leveraging the `csv` module in the python standard library here.  It'll make it easier to optionally adjust the delimiter, and I think it'll cut the number of lines of code required in half (since you don't need to explicitly provide the delimiter after every element).